client/rest-high-level/src/main/java/org/elasticsearch/client/indices/ResizeRequest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.action.admin.indices.alias.Alias;
+import org.elasticsearch.action.support.ActiveShardCount;
+import org.elasticsearch.client.TimedRequest;
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.client.ValidationException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Request class to resize an index
+ */
+public class ResizeRequest extends TimedRequest implements Validatable, ToXContentObject {
+
+    private ActiveShardCount waitForActiveShards;
+    private final String sourceIndex;
+    private final String targetIndex;
+    private Settings settings = Settings.EMPTY;
+    private Set<Alias> aliases = new HashSet<>();
+
+    public ResizeRequest(String targetIndex, String sourceIndex) {
+        this.targetIndex = targetIndex;
+        this.sourceIndex = sourceIndex;
+    }
+
+    public ResizeRequest setSettings(Settings settings) {
+        this.settings = settings;
+        return this;
+    }
+
+    public Settings getSettings() {
+        return this.settings;
+    }
+
+    public ResizeRequest addAlias(Alias alias) {
+        this.aliases.add(alias);
+        return this;
+    }
+
+    public ResizeRequest addAliases(List<Alias> aliases) {
```

Comment:
id maybe change this to setAliases() since there is no way to "clear" the aliases list if youve somehow messed up w/o new'ing it again